src/server/pkg/migrations/migrations.go

```diff
@@ -68,28 +68,44 @@ func InitialState() State {
 	return State{
 		name: "init",
 		change: func(ctx context.Context, env Env) error {
-			_, err := env.Tx.ExecContext(ctx, `CREATE TABLE IF NOT EXISTS migrations (
-				id BIGINT PRIMARY KEY,
-				NAME VARCHAR(250) NOT NULL,
-				start_time TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
-				end_time TIMESTAMP
-			);
-			INSERT INTO migrations (id, name, start_time, end_time) VALUES (0, 'init', CURRENT_TIMESTAMP, CURRENT_TIMESTAMP) ON CONFLICT DO NOTHING;
-			`)
-			return err
+			return nil
 		},
 	}
 }
 
 // ApplyMigrations does the necessary work to actualize state.
 // It will manipulate the objects available in baseEnv, and use the migrations table in db.
 func ApplyMigrations(ctx context.Context, db *sqlx.DB, baseEnv Env, state State) error {
+	tx, err := db.BeginTxx(ctx, &sql.TxOptions{})
```

Comment:
Previously we ran each migration step in it's own transaction. This runs this risk of leaving a customer database in a weird in-between state where they can't run a previous release or the current release. It also meant we needed to re-acquire the lock on the migrations table for each step in the migration, which would make for confusing logs in pachd if multiple instances were competing to make progress.

Reply:
More complicated migrations involving things other than the database might benefit from having checkpoints.  We can't do that if everything rolls back.  If we are shuffling things around in object storage or doing ETL to populate a table we would probably do it in steps to avoid losing hours of work.

Confusing logs, but otherwise correct behavior doesn't seem like an issue considering this is a transition step to something better like a single container executing the migration while everyone blocks.

This seems orthogonal to rollback, which can also be done state by state.  And I totally agree that we need a way to rollback.